pytorch_lightning/utilities/device_parser.py

```diff
@@ -107,13 +112,25 @@ def parse_tpu_cores(tpu_cores: Union[int, str, List]) -> Optional[Union[List[int
 
 
 def _normalize_parse_gpu_string_input(s: Union[int, str, List[int]]) -> Union[int, List[int]]:
-    if isinstance(s, str):
-        if s == '-1':
-            return -1
-        else:
-            return [int(x.strip()) for x in s.split(',') if len(x) > 0]
-    else:
+    if not isinstance(s, str):
         return s
+    if s == '-1':
+        return -1
+    elif ',' in s:
+        return [int(x.strip()) for x in s.split(',') if len(x) > 0]
+    else:
+        num_gpus = int(s.strip())
+        if _compare_version("pytorch_lightning", operator.lt, "1.5"):
+            # TODO: remove warning in v1.5 and update docs in docs/advanced/multi-gpu.rst regarding GPU selection
+            rank_zero_warn(
+                f"Parsing of the Trainer argument gpus='{s}' (string) will change in the future."
+                f" In the current version of Lightning, this will select"
+                f" CUDA device with index {num_gpus}, but from v1.5 it will select gpus"
+                f" {list(range(num_gpus))}.",
+                DeprecationWarning,
+            )
+            return [num_gpus]
+        return num_gpus
```

Comment:
@awaelchli I wonder what values reach this point.

Reply:
The last return statement? Currently, this cannot be reached, because we are below version 1.5.
But from version 1.5 it will hit the last return, i.e., `Trainer(gpus="3")` will be interpreted as the number of gpus, just like `Trainer(gpus=int)` does.

Reply:
I did it this way so that in 1.5 all we have to do is delete the if block above and everything else will be correct.

Reply:
Sounds good, thanks!